pkg/controller/service/service_controller.go

```diff
@@ -280,13 +280,13 @@ func (s *ServiceController) createLoadBalancerIfNeeded(key string, service *v1.S
 
 		// TODO: We could do a dry-run here if wanted to avoid the spurious cloud-calls & events when we restart
 
-		// The load balancer doesn't exist yet, so create it.
-		s.eventRecorder.Event(service, v1.EventTypeNormal, "CreatingLoadBalancer", "Creating load balancer")
-		newState, err = s.createLoadBalancer(service)
+		// Ensure existence of the load balancer
```

Comment:
Probably just remove this comment? Also other than ensures LB exist, ensureLoadBalancer() could also ensure LB updated :)